### PR TITLE
Update outdated npm packages to latest

### PR DESCRIPTION
**Pull request by carlosperez**
Since bitHound started scoring dependencies, we noticed a dramatic drop in score for the Falcor project. This was caused by several outdated npm packages. We identified the packages that we thought were safe to update to latest in `package.json`. After an npm install, tests and build both still pass.

- [x] Update `package.json` (see commit for details)
- [x] All tests pass
- [x] Build ... builds. ;)

A few notes:
- The update includes 1 top level package (promise) and the others are all dev dependencies.
- The update to promise simple changes it from `7.0.0.` to `^7.0.4`
- As we are not intimately familiar with the project, more eyes are needed on this to ensure everything works as expected.
- We did not update packages that were related to the test suite (Chai, Mocha, etc.) as doing so prevented tests from even running.

See justification for this update these changes here https://www.bithound.io/github/bithound/falcor/npm.updates

### CODE REVIEW

**Comment by jhusain** [comment]
@sdesai do you mind taking a look at this?

**Comment by sdesai** [comment]
Thanks @carlosperez

We'll take a look at this, to make sure the updates aren't breaking anything before merging in. Since it's not breaking any functionality with the current version of modules, will look at this once we've made our way through some of the other open issues.

**Comment by carlosperez** [comment]
Makes sense! Thanks @sdesai

File: package.json [hunk 1]
```diff
@@ -70,18 +70,18 @@
     "karma-benchmark": "^0.4.0",
     "karma-chrome-launcher": "^0.2.0",
     "karma-firefox-launcher": "^0.1.6",
-    "karma-junit-reporter": "^0.2.2",
+    "karma-junit-reporter": "^0.3.3",
     "karma-safari-launcher": "^0.1.1",
-    "lodash": "~2.4.1",
-    "minifyify": "7.0.0",
+    "lodash": "~3.10.1",
```
Comment (megawac) [inline package.json]: why are you patch locking this one and not underscore?

Comment (carlosperez) [inline package.json]: That was a mistake on my part, I intended to keep to npm's default of using the caret. However, given that there are already different range rules being used, is there a preferred rule going forward?

Comment (sdesai) [inline package.json]: The difference is likely due to the shift in --save, --save-dev behavior based on the version of npm various developers were using, as opposed to anything conscious.

I'm thinking ^ (flexible minor version) is fine for devDeps (and we'd have to react ASAP to breakage if deps sneak in breaking changes with minor versions), but for product deps, we want to stay with fixed versions.

**Comment by carlosperez** [comment]
@sdesai I switched lodash and through2 to use the caret convention and version locked promise at `7.0.4`. Let me know if you want me to rebase these changes into a single commit for a cleaner commit tree, or if this is fine as is. 

Tests and build still working after these changes.

**Comment by sdesai** [comment]
@carlosperez - No that's fine, don't worry about it.

**Comment by sdesai** [comment]
Hey Carlos, sorry about the delay, just been focused on cleaned up some other functional bugs before introducing this flux.

I've issued a new PR, to replace this one, since I took the opportunity to remove some of our unused deps. Closing this one out, in favor of : https://github.com/Netflix/falcor/pull/536

**Comment by carlosperez** [comment]
Hi @sdesai – no worries! Great to see them getting updated!

**Comment by sdesai** [comment]
I can open this conversation on bithound, but as a bithound feature, it would be useful to be able to separate code complexity/lint results from dependency analysis. We may not always update dependencies just because there's a new major release available (that's a function of resourcing/cost to migrate vs. value/benefit), but it would be nice to be able to bubble up fluctuations in the lint and complexity number build over build, without it being influenced by dependencies.

**Comment by carlosperez** [comment]
Hi @sdesai – any chance I can get in touch with you directly to follow up on this? Just have some questions.

**Comment by sdesai** [comment]
Yep, absolutely. I'll ping you.

**Pull request closed without merging**